BUILD.gn

```diff
@@ -11,6 +11,12 @@ group("flutter") {
     "//flutter/sky",
   ]
 
+  if (current_toolchain == host_toolchain) {
+    deps += [
+        "//dart:create_sdk"
```

Comment:
In the Fuchsia build, it's not for this BUILD.gn file to decide whether we want to build the Dart SDK.  That's the responsibility of another part of the system (probably something in the build layer).  Think about it this way: if Fuchsia decided to switch to a prebuilt Dart SDK (e.g., similar to how Fuchsia uses a prebuilt llvm toolchain), we shouldn't need to modify this file.